binary_transparency/firmware/integration/ft_test.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+set -e
+. $(go list -f '{{.Dir}}' github.com/google/trillian)/integration/functions.sh
+INTEGRATION_DIR="$( cd "$( dirname "$0" )" && pwd )"
+. "${INTEGRATION_DIR}"/ft_functions.sh
+
+ft_prep_test
+
+function cleanup {
+banner "Cleaning up"
+ft_stop_test
+TO_KILL=()
+}
+
+trap cleanup EXIT
+
+DEVICE_STATE=$(mktemp -d /tmp/dummy-XXXXX)
+UPDATE_FILE=$(mktemp /tmp/update-XXXXX.json)
+
+
+# Cleanup for the Trillian components
+TO_DELETE="${TO_DELETE} ${ETCD_DB_DIR}"
+TO_KILL+=(${LOG_SIGNER_PIDS[@]})
+TO_KILL+=(${RPC_SERVER_PIDS[@]})
+TO_KILL+=(${ETCD_PID})
+
+# Cleanup for the personality
+TO_DELETE="${TO_DELETE} ${FT_CAS_DB} ${DEVICE_STATE} ${UPDATE_FILE} ${FT_MONITOR_LOG}"
+TO_KILL+=(${FT_SERVER_PID})
+TO_KILL+=(${FT_MONITOR_PID})
+
+echo "Running test(s)"
+pushd "${INTEGRATION_DIR}"
+
+banner "Logging initial firmware"
+go run ../cmd/publisher/ \
+    --log_url=http://${FT_SERVER} \
+    --binary_path ../testdata/firmware/dummy_device/example.wasm  \
+    --revision 1 \
+    --output_path=${UPDATE_FILE} \
+    --logtostderr -v 2
+
+banner "Force flashing device (init)"
+go run ../cmd/flash_tool/ \
+    --log_url=http://${FT_SERVER} \
+    --update_file=${UPDATE_FILE} \
+    --dummy_storage_dir=${DEVICE_STATE} \
+    --force \
+    --v 2 --logtostderr
+
+banner "Booting device with initial firmware"
+go run ../cmd/emulator/dummy/ \
+    --dummy_storage_dir=${DEVICE_STATE} \
+    --logtostderr
+
+banner "Logging update firmware"
+go run ../cmd/publisher/ \
+    --log_url=http://${FT_SERVER} \
+    --binary_path ../testdata/firmware/dummy_device/example.wasm  \
+    --revision 2 \
+    --output_path=${UPDATE_FILE} \
+    --logtostderr -v 2
+
+banner "Force flashing device (update)"
+go run ../cmd/flash_tool/ \
```

Comment:
This doesn't have the `log_url` provided?

Reply:
In general, should we be asserting the return code of these tools?

Reply:
I believe we are - there's a `set -e` up at the top.
At some point soon we'll want to assert a non-zero return, but we'll cross that bridge when we get to it :)

Reply:
(good spot on the log URL - updated)

Reply:
I'm unclear why this didn't hard fail on this command when the log_url wasn't provided. I would have expected it to have tried getting the firmware from a different log than previously and it should have failed?

Reply:
Seems to work - just pushed a commit which sets the `log_url` incorrectly and the integration test logged this:
```
---[Logging update firmware]----------------------------------------------
F1124 12:18:50.877536   13917 publish.go:73] Failed to get a pre-submission checkpoint from log: "Get \"http://localhost:1234/ft/v0/get-root\": dial tcp 127.0.0.1:1234: connect: connection refused"
exit status 1
```

and Travis barfed.